persistence/sql/persister_identity.go

```diff
@@ -39,44 +43,71 @@ func (p *Persister) ListVerifiableAddresses(ctx context.Context, page, itemsPerP
 }
 
 func (p *Persister) ListRecoveryAddresses(ctx context.Context, page, itemsPerPage int) (a []identity.RecoveryAddress, err error) {
+	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.ListRecoveryAddresses")
+	defer span.End()
+
 	if err := p.GetConnection(ctx).Where("nid = ?", corp.ContextualizeNID(ctx, p.nid)).Order("id DESC").Paginate(page, x.MaxItemsPerPage(itemsPerPage)).All(&a); err != nil {
 		return nil, sqlcon.HandleError(err)
 	}
 
 	return a, err
 }
 
-func (p *Persister) FindByCredentialsIdentifier(ctx context.Context, ct identity.CredentialsType, match string) (*identity.Identity, *identity.Credentials, error) {
-	nid := corp.ContextualizeNID(ctx, p.nid)
+func stringToLowerTrim(match string) string {
+	return strings.ToLower(strings.TrimSpace(match))
+}
 
-	var cts []identity.CredentialsTypeTable
-	if err := p.GetConnection(ctx).All(&cts); err != nil {
-		return nil, nil, sqlcon.HandleError(err)
+func (p *Persister) normalizeIdentifier(ctx context.Context, ct identity.CredentialsType, match string) string {
+	switch ct {
+	case identity.CredentialsTypeLookup:
+		// lookup credentials are case-sensitive
+		return match
+	case identity.CredentialsTypeTOTP:
+		// totp credentials are case-sensitive
+		return match
+	case identity.CredentialsTypeOIDC:
+		// OIDC credentials are case-sensitive
+		return match
+	case identity.CredentialsTypePassword:
+		// fandom-start - temporary fix during identifier migration
```

Comment:
Do we still need this? If the migration is done, maybe we can remove the case-sensitive related changes.